appgate/syncer/git.py

```diff
@@ -131,35 +146,101 @@ def push_change(self, branch: str) -> None:
     def create_or_update_pull_request(
         self,
         branch: str,
-        pull_request: PullRequest | None,
+        pull_request: int | None,
         commits: Dict[str, List[Tuple[str, GitCommitState]]],
-    ) -> PullRequest | None:
-        pass
+    ) -> int | None:
+        raise NotImplementedError()
 
+    def checkout(
```

Comment:
I like the idea of making generic the logic business here but this should be implemented in `checkout_branch` I think. Not need to wrap this `checkout` method around.